spring-boot-project/spring-boot-tools/spring-boot-buildpack-platform/src/main/java/org/springframework/boot/buildpack/platform/docker/LogUpdateEvent.java

```diff
@@ -90,7 +90,15 @@ private static LogUpdateEvent read(InputStream inputStream) throws IOException {
 		if (header == null) {
 			return null;
 		}
-		StreamType streamType = StreamType.values()[header[0]];
+
+		// First byte denotes stream type. 0 = stdin, 1 = stdout, 2 = stderr
+		byte streamTypeId = header[0];
+		if (streamTypeId < 0 || streamTypeId >= StreamType.values().length) {
+			// Stream type is out of bounds, ignore this event. See gh-29307
```

Comment:
I thought about logging a warning here, but I didn't found any logging infrastructure. Any thoughts on that?

Reply:
It's not particularly easy. There's a `BuildLog` interface that's used to do all the logging. I think Gradle outputs it to a `PrintStream` and maven has a `MojoBuildLog` class that send the text to the standard maven logger. Probably the easiest option here is to slightly abuse the `LogUpdateEvent`. I think you could do perhaps throw an exception from `read` and catch it in `readAll` then construct a `LogEvent` with a sensible message and pass that to the `consumer` before exiting.

Reply:
Will do that, thanks!